README.md

```diff
@@ -7,185 +7,181 @@
+In the following, the complete process of specifying a message, generating code, and using the generated code is demonstrated using a small example.
+
+### Specification
+
+The following sample specification describes a protocol `TLV` with one message type `Message` consisting of three fields:
+
+- A field `Tag` of 2 bit length,
+- a field `Value_Length` of 14 bit length, and
+- a field `Value`, whose length is specified by the value in `Value_Length`.
+
+The `Tag` can have two valid values: `1` (`Msg_Data`) and `3` (`Msg_Error`). In case `Tag` has a value of `1` the fields `Value_Length` and `Value` follow. `Message` contains only one field, if the value of `Tag` is `3`. All other values of `Tag` lead to an invalid message.
 
 ```
-package Foo is
+package TLV is
 
    type Tag_Type is (Msg_Data => 1, Msg_Error => 3) with Size => 2;
    type Length_Type is mod 2**14;
 
-   type Bar is
+   type Message is
       message
-         Tag          : Tag_Type
-            then Value_Length
+         Tag    : Tag_Type
+            then Length
                if Tag = Msg_Data,
             then null
                if Tag = Msg_Error;
-         Value_Length : Length_Type
+         Length : Length_Type
             then Value
-               with Length = Value_Length;
-         Value        : Payload_Type;
+               with Length => Length * 8;
+         Value  : Payload_Type;
        end message;
 
-end Foo;
+end TLV;
 ```
 
-## Code Generation
-
-The code generator is able to generate dissector code based on a set of protocol specifications. The generated code allows to validate and parse protocol messages and thereby respects all specified restrictions in and between protocols. Adding the generation of messages is in planning. By using [SPARK](https://www.adacore.com/about-spark) we are able to prove the absence of runtime errors and prevent the incorrect usage of the generated code (e.g., enforce that a field of a protocol message is validated before accessed).
-
-The code generator creates a number of packages for a protocol specification. All basic types like integers, enumerations and arrays are collectively declared in one package. For each message type a child package is generated which contains validation and access functions for each field of the message.
-
-A user of the generated code has to validate a protocol field or the whole protocol message before accessing the data of a particular protocol field. The SPARK toolset in combination with the generated verification conditions make it possible to ensure this property, and so prevent incorrect usage.
+### Generating Code
 
-### Example
-
-The file `foo.ads` contains all basic types of the package `Foo`:
+With the sub-command `check` the correctness of the given specification file can be verified.
```

Comment:
verified -> checked (we don't want this to be confused with formal verification)